pkg/resource/plugin/plugin.go

```diff
@@ -45,6 +45,7 @@ type plugin struct {
 
 	Bin    string
 	Args   []string
+	Env    []string
```

Comment:
If the expectation that `Env` contains key value pairs of the form "X=Y", then that definitely should be called out. e.g. what happens when encountering newlines in the config value? Will that work? Or does "Y" need to be enclosed in quotation marks? etc.